### ISSUE TITLE
How to train mask rcnn on large images 4800x5900?

### ISSUE THREAD

**Issue by PrinceB7** [matterport/Mask_RCNN#2717]
Hello everyone,

How can I train the model on large images? The problem is some of the objects are too small compared to the image size, thus I cannot resize the whole image. On average, I have 3-4 objects per image. Even though I trained for a large number of epochs and used many samples, Mask RCNN is just ignoring small objects....

Can anyone help me with this? any insights would be helpful

Thanks.

**Comment by jqx37** [comment]
> Hello everyone,
> 
> How can I train the model on large images? The problem is some of the objects are too small compared to the image size, thus I cannot resize the whole image. On average, I have 3-4 objects per image. Even though I trained for a large number of epochs and used many samples, Mask RCNN is just ignoring small objects....
> 
> Can anyone help me with this? any insights would be helpful
> 
> Thanks.
To train such a large image, you may need a GPU that has a very large VRAM. What GPU are you using?

**Comment by PrinceB7** [comment]
@jqx37 
GPU is not a problem. Because training is going well. I trained for 300 epochs without memory problems. Now i am using 2 Tesla V100s (32GB each). I can use 4 GPUs if need be. 

the problem is that the accuracy for very small objects (e..g. 40x70) is low. I assume when the image is downsampled, the model can no longer see these objects? 

Because if the image size is like 1920x1080 and the object bbox is 40x70, the model doesn’t seem to have this problem. But if I use 4800x5900, the accuracy drops drastically. Good news is that it can still detect larger objects in the image :)

how to handle such object-image size ratios? 

thanks.

**Comment by jqx37** [comment]
@PrinceB7 Hi. Those are fairly small objects compare to the size of the image. Is that ok to use the full size image instead of the down-sampled ones?

**Comment by jqx37** [comment]
@PrinceB7 Also, IMO the dataset size is also a factor that affects the results on segmenting such small objects, since those objects have too little "influence" compare to other parts in the image during the training process.

**Comment by PrinceB7** [comment]
> @PrinceB7 Hi. Those are fairly small objects compare to the size of the image. Is that ok to use the full size image instead of the down-sampled ones?

@jqx37 yes, this is what i want to know. Because I don’t think the model feeds 4800x5900 images, right? I am not downsampling the images but I suspect the model does. 

if it isn’t downsampled, i guess the accuracy improves.

when it comes to dataset size, i think it is alright. also, other objects are detected really well, Except these small ones.

**Comment by jqx37** [comment]
@PrinceB7 If you are using the original repo, I think the images would be padded to 5900 by 5900 (if square padding). I guess other objects are much bigger than those troublesome ones, is that right? If so, my understanding is the smaller objects have to rise the "attention" of your model. I think either a larger dataset that have those small objects may help, or "emphasizing" the loss on those objects may work? Not sure about this though.

**Comment by PrinceB7** [comment]
@jqx37 yes right, mate. this is what i think too. 
Good, if model uses 5900x5900 then one of my problems is gone :) I was’t sure about that until u clarified. 

I think i should use slicing then. It will take little bit of human labor to re-annotate but i hope it helps.

thank u!

**Comment by jqx37** [comment]
@PrinceB7 You are welcome. Maybe you can try to change the class loss weight, which can penalize the class loss a bit more. Personally it did not work as well as I expected, but I think I did not have a chance to try more models, that might need to be taken into consideration. Anyway, good luck, may you get a better result.

**Comment by jqx37** [comment]
@PrinceB7 Hi, how did you set up your GPU to share the memory of your GPUs? I have two A6000 with NVLink, and I have problems using the sharing the VRAM. Thanks.
